### PR TITLE
Basic process mem usage functions 

### PR DESCRIPTION
**Pull request by DmitriyMusatkin**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/common/system_resource_util.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#ifndef AWS_COMMON_SYSTEM_RESOURCE_UTIL_H
```
Comment (waahm7) [inline include/aws/common/system_resource_util.h:1]: Debatable: We already have a system_info header which might be a good place for these functions.

File: CMakeLists.txt [hunk 2]
```diff
@@ -82,7 +82,8 @@ if (WIN32)
     endif ()
 
     list(APPEND PLATFORM_DEFINES WINDOWS_KERNEL_LIB=${WINDOWS_KERNEL_LIB})
-    list(APPEND PLATFORM_LIBS BCrypt ${WINDOWS_KERNEL_LIB} Ws2_32 Shlwapi)
+    list(APPEND PLATFORM_DEFINES PSAPI_VERSION=1)
```
Comment (waahm7) [inline CMakeLists.txt:85]:
```suggestion
    /* PSAPI_VERSION=1 is required for GetProcessMemoryInfo */
    list(APPEND PLATFORM_DEFINES PSAPI_VERSION=1)
```

File: tests/system_resource_util_test.c [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    struct aws_memory_usage mu;
+    ASSERT_SUCCESS(aws_init_memory_usage_for_current_process(&mu));
+
+    ASSERT_TRUE(mu.maxrss > 0);
+
+    return 0;
+}
+
+AWS_TEST_CASE(test_memory_usage_maxrss, s_test_memory_usage_maxrss)
```
Comment (waahm7) [inline tests/system_resource_util_test.c:22]: trivial: missing blank line at EOF.

**Review by waahm7 — APPROVED** [review]

File: include/aws/common/system_resource_util.h (hunk 1 above)
Comment (DmitriyMusatkin) [reply include/aws/common/system_resource_util.h:1]: Im on the fence about it. I was originally planning to put it into system_info, but functions here have a slightly different use case - how many resource is my process consuming vs which system im running on. I was also planning to add functions to modify resource limits to this file, which will diverge things even more in future. So im leaning towards a separate header.

**Pull request merged**